### ISSUE TITLE
`zkevm_getBatchByNumber` change  batchl2data between calls

### ISSUE THREAD

**Issue by joanestebanr** [0xPolygon/cdk-erigon#822]
#### System information

Erigon version: `1.5.9`

Chain/Network: `cardona`

#### Expected behaviour
When calling the end-point `zkevm_getBatchByNumber` of the batch in progress the batchl2data must be incremental. It shrink in some cases, check next log: 
```
URL= https://rpc-erigon.zkevm-testnet.com/
last batch:  105404
initial len=1292
increased: 18 total_new_len: 1310
===increased: 18 total_new_len: 1328
====increased: 18 total_new_len: 1346
===increased: 18 total_new_len: 1364
===increased: 18 total_new_len: 1382
===increased: 18 total_new_len: 1400
====increased: 242 total_new_len: 1642
previous len=1642 > current=1640
error data
```
Another cases is the previous data changes: 
```
URL= https://rpc-erigon.zkevm-testnet.com/
last batch:  105404
initial len=2166
==increased: 18 total_new_len: 2184
===increased: 18 total_new_len: 2202
===increased: 18 total_new_len: 2220
====increased: 18 total_new_len: 2238
===increased: 18 total_new_len: 2256
====increased: 18 total_new_len: 2274
===increased: 18 total_new_len: 2292
====increased: 18 total_new_len: 2310
===increased: 18 total_new_len: 2328
====increased: 18 total_new_len: 2346
===increased: 18 total_new_len: 2364
===increased: 18 total_new_len: 2382
===increased: 18 total_new_len: 2400
====increased: 18 total_new_len: 2418
====increased: 18 total_new_len: 2436
===increased: 18 total_new_len: 2454
===increased: 18 total_new_len: 2472
====increased: 18 total_new_len: 2490
==
=increased: 18 total_new_len: 2508
=======increased: 18 total_new_len: 2526
==increased: 2934 total_new_len: 5460
previous contents doesnt match new content
0x0b00000003000000000b00000003000000000b00000003000000000b0000000300000000ef8303e2288501bd3f278082520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a8080b7d38f4b8fa4b0996932f7aa4295f6f8c7db8482e0ae14c3a1431fc23a3e41c96e100a1d47c9dc41c87296bd7f4f435855b247bae6f8393b25e04c87d4fae6761c150b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000ee8303e22984946a628082520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a8080e4eec747189ee792374ecd0b52c38f5abefc2a74f1f80d8667ff42ede9fb4f33425658045f6b1a8d2fe334062163cb2bfe6bf08d8a039ccc56168236d8dbd95d1c150b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000004000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000ef8303e22a85019ed92c0082520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a80805edd734781e58a5e73c6813c9d51a0f43f9c1dd2ad07fc3f091587ce82dacdc721ed1db166bfa2179ef5746a61b1fa9eaee4668d3f8328d191fa10fc02856baf1b150b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000ec8303e22b82039282520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a808042f3256cb125993ba6e1019c1d1619010aed56fcfda52faceb39b7c0505b82e85a3f3be9616773388075801135b054a5d6b2e5d4dae2e82739c3b787922c4c3a1b150b00000003000000000b00000004000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000f9056d825fa783068456830f7220948af367b161fa897fa7e89262c9b4ecb8bc0d7d5680b9054457ffc86100000000000000000000000000000000000000000000000000000000000001200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000043d027ac5a7a568fc5e1dd709c788ef5c85129553908a28d5a0c23181f771357e30c5baaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa0000000000000000000000000000000000000000000000000000000000000001012ebe915aca3ce79a5d8b9565e00f511719d747cfc8c649a8f33156a4ea6317294bbf8f23786064b9a5b14d7f4041656228fdef028efaf3f13d74d1370504f10000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000180000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000010000000000000000000000004e70004b01987b4fd7dacab348f0b39e9ae2611000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000290000000000000000000000000000000000000000000000000ed01020ccd14d8000000000000000000000000000000000000000000000000000000000669a45d900000000000000000000000000000000000000000000000000000000000000019f7483837c7d492f6edb1e77d45837c6f3e4a9bedb76db2c2d6dba54a463b4ad000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000180000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000020000000000000000000000004e70004b01987b4fd7dacab348f0b39e9ae26110000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000002a0000000000000000000000000000000000000000000000000f2a05a197c59b0000000000000000000000000000000000000000000000000000000000669a45d90000000000000000000000000000000000000000000000000000000000000001d7203ceabc1ef4122a2d7c6dec1c7eca490e20ed09b28aeb0c17e7f5c4008b390000000000000000000000000000000000000000000000000000000000000002ff0100000000000000000000000000000000000000000000000000000000000082098a8080e2ed549bee24376e880fd8c4b631d20db49223b1fc0d25ee0eabbb05329b2f1e2a9aa758948008c9fb1c0f1b06d6f28bff4894d32f6cb064125a9b26ccca2f491b2d
0x0b00000003000000000b00000003000000000b00000003000000000b0000000300000000ef8303e2288501bd3f278082520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a8080b7d38f4b8fa4b0996932f7aa4295f6f8c7db8482e0ae14c3a1431fc23a3e41c96e100a1d47c9dc41c87296bd7f4f435855b247bae6f8393b25e04c87d4fae6761c150b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000ee8303e22984946a628082520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a8080e4eec747189ee792374ecd0b52c38f5abefc2a74f1f80d8667ff42ede9fb4f33425658045f6b1a8d2fe334062163cb2bfe6bf08d8a039ccc56168236d8dbd95d1c150b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000004000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000ef8303e22a85019ed92c0082520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a80805edd734781e58a5e73c6813c9d51a0f43f9c1dd2ad07fc3f091587ce82dacdc721ed1db166bfa2179ef5746a61b1fa9eaee4668d3f8328d191fa10fc02856baf1b150b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000ec8303e22b82039282520894417a7ba2d8d0060ae6c54fd098590db854b9c1d58609184e72a0008082098a808042f3256cb125993ba6e1019c1d1619010aed56fcfda52faceb39b7c0505b82e85a3f3be9616773388075801135b054a5d6b2e5d4dae2e82739c3b787922c4c3a1b150b00000003000000000b00000004000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b00000003000000000b0000000300000000f9056d825fa7832444aa830f7220948af367b161fa897fa7e89262c9b4ecb8bc0d7d5680b9054457ffc86100000000000000000000000000000000000000000000000000000000000001200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000043d027ac5a7a568fc5e1dd709c788ef5c85129553908a28d5a0c23181f771357e30c5baaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa0000000000000000000000000000000000000000000000000000000000000001012ebe915aca3ce79a5d8b9565e00f511719d747cfc8c649a8f33156a4ea6317294bbf8f23786064b9a5b14d7f4041656228fdef028efaf3f13d74d1370504f10000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000180000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000010000000000000000000000004e70004b01987b4fd7dacab348f0b39e9ae2611000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000290000000000000000000000000000000000000000000000000ed01020ccd14d8000000000000000000000000000000000000000000000000000000000669a45d900000000000000000000000000000000000000000000000000000000000000019f7483837c7d492f6edb1e77d45837c6f3e4a9bedb76db2c2d6dba54a463b4ad000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000180000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000020000000000000000000000004e70004b01987b4fd7dacab348f0b39e9ae26110000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000002a0000000000000000000000000000000000000000000000000f2a05a197c59b0000000000000000000000000000000000000000000000000000000000669a45d90000000000000000000000000000000000000000000000000000000000000001d7203ceabc1ef4122a2d7c6dec1c7eca490e20ed09b28aeb0c17e7f5c4008b390000000000000000000000000000000000000000000000000000000000000002ff0100000000000000000000000000000000000000000000000000000000000082098a8080e2ed549bee24376e880fd8c4b631d20db49223b1fc0d25ee0eabbb05329b2f1e2a9aa758948008c9fb1c0f1b06d6f28bff4894d32f6cb064125a9b26ccca2f491b2d
error data
end for batch  105404

```

#### Actual behaviour
Calling the end-point sometimes the batchl2data from previous call doesn't match next one.

#### Steps to reproduce the behaviour
You can reproduce with that script (set `rpc_url` with the  RPC under test): 
```
import requests
import json

# Variables
rpc_url = "https://rpc.cardona.zkevm-rpc.com/"  # Change for the RPC UL
key_batchL2Data = "batchL2Data"
key_closed = "closed"
def get_lastBatchNum():
    payload = {
        "method": "zkevm_batchNumber",
        "params": [],
        "id": 1,
        "jsonrpc": "2.0"
    }
    headers = {
        "Content-Type": "application/json"
    }
    response = requests.post(rpc_url, headers=headers, data=json.dumps(payload))
    return int(response.json()['result'],16)

def get_batch(batch_num):
    batch_num_hex = hex(batch_num) # Sustituye VALOR_DEL_BATCH_NUM_HEX por el valor de $BATCH_NUM_HEX

    # Datos de la solicitud
    payload = {
        "method": "zkevm_getBatchByNumber",
        "params": [batch_num_hex],
        "id": 1,
        "jsonrpc": "2.0"
    }

    # Cabeceras de la solicitud
    headers = {
        "Content-Type": "application/json"
    }

    # Realizar la solicitud POST
    response = requests.post(rpc_url, headers=headers, data=json.dumps(payload))

    # Imprimir la respuesta
    #print(response.json())
    return response.json()['result']

def compare_data(num_request,old, new) -> bool:
    if old == new:
        if num_request%80 == 0:
            print(f"=")
        else:
            print(f"=", end='')
        return True
    if len(old) > len(new):
        print(f"previous len={len(old)} > current={len(new)}")
        return False
    common_substr = new[0:len(old)]
    if len(common_substr) != len(old):
        print("script error substr is not correct: ", len(common_substr), " ", len(old))
        print(common_substr)
        print(old)
        return False
    if common_substr != old:
        print(f"previous contents doesnt match new content")
        print(common_substr)
        print(old)
        return False
    print(f"increased: {len(new)-len(old)} total_new_len: {len(new)}")
    return True



wip_batch_number= get_lastBatchNum()
print("last batch: ", wip_batch_number)
initial = get_batch(wip_batch_number)
current_data = initial[key_batchL2Data]
print(f"initial len={len(current_data)}")
num_request = 0
while True:
    num_request+=1
    current = get_batch(wip_batch_number)
    if not  compare_data(num_request,current_data, current[key_batchL2Data]):
        print("error data")
        break

    current_data = current[key_batchL2Data]
    #print(f".", end='')
    if current[key_closed] == True:
        old_batch_number = wip_batch_number
        wip_batch_number= get_lastBatchNum()
        if old_batch_number == wip_batch_number:
            print(f"\nBatch {old_batch_number} is closed. No new Batch")
        else:
            initial = get_batch(wip_batch_number)
            current_data = initial[key_batchL2Data]
            print(f"\nBatch {old_batch_number} is closed. new WIP Batch {wip_batch_number} initial len={len(current_data)}")


print("end for batch ",wip_batch_number)
```

#### Backtrace

````
[backtrace]
````

**Comment by Sharonbc01** [comment]
Is [#809](https://github.com/0xPolygonHermez/cdk-erigon/issues/809) related to this issue?

**Comment by mandrigin** [comment]
Three bytes that are different, starting from character indexed 2558

`068456`
`2444aa`

```
Index: 2558 v1: 0 v2: 2
Index: 2559 v1: 6 v2: 4
Index: 2560 v1: 8 v2: 4
Index: 2561 v1: 4 v2: 4
Index: 2562 v1: 5 v2: a
Index: 2563 v1: 6 v2: a
```

**Comment by Sharonbc01** [comment]
Appears resolved Ideally like to have12 hrs syncing with no issues.

cc: @mandrigin @eduadiez

**Comment by Sharonbc01** [comment]
Update shared by @rafael-genes showing the error persists on the rpc node.

I clone the rpc-debug on the rest of the rpc instances.
I scale up to 3 instances the main rpc.
I resync both of our permissionless and the error persist on both.
I'm doing last checks before detailing on the group, but seems like the issue is not fixed and there's a hidden reason that corrupts the datadirs of the main RPC

**Comment by Sharonbc01** [comment]
No errors for > 12 hrs so agreed we can close.💃

**Comment by Sharonbc01** [comment]
Reopening as after RPC v1.2.15.3was deployed to mainnet partners had issues with legacy nodes.

Reverted to legacy and investigating
Currently what we found that even under the LB L2BatchData for some batches differs if you call it multiple times. That doesn’t happen if there is a single Erigon node.

**Comment by mandrigin** [comment]
That is fixed by 1.2.15.4 -- gas price issue. So far no issues with sync since 1.2.15.4 deployed. Closing this one.
